src/mongoose_hooks.erl

```diff
@@ -1044,6 +1046,14 @@ mam_flush_messages(HookServer, MessageCount) ->
 mam_archive_sync(HostType) ->
     run_hook_for_host_type(mam_archive_sync, HostType, ok, [HostType]).
 
+%% @doc Notifies of a message retraction
+-spec mam_retraction(mongooseim:host_type(),
+                     mod_mam_utils:retraction_info(),
+                     mod_mam_rdbms_arch:env_vars()) ->
+    mod_mam_utils:retraction_info().
+mam_retraction(HostType, RetractionInfo, Env) ->
+    {ok, RetractionInfo1} = gen_hook:run_fold(mam_retraction, HostType, RetractionInfo, Env),
```

Comment:
There is a helper for calling hooks at the bottom of this file. It also handles the `ok/stop` matching (discarding both actually).

Reply:
I'd use the helper and revert the change for returning `{stop, RetVal}` in case of a stopped hook. We may not use it now, but at least it's not surprising that it's getting changed to an `ok` if we ever would want to...

Reply:
Oh, that helper 😄